validphys2/src/validphys/results.py

```diff
@@ -48,11 +48,13 @@ class NNPDFDataResult(Result):
     """A result fills its values from a pandas dataframe
     For legacy (libNNPDF) compatibility, falls back to libNNPDF attributes"""
 
-    def __init__(self, dataobj=None, central_value=None):
+    def __init__(self, dataobj):
         # This class is used by both validphys and libNNPDF objects
-        # when central_value is not explictly passed, fallback to
-        # libNNPDF object .get_cv()
-        if central_value is None:
+        # At this point only the result of the ThPredictions is a vp object
+        # which includes a special CV column which needs to be pop'd
+        try:
+            central_value = dataobj.pop("CV")
```

Comment:
Maybe mutating the object like this will be dangerous/confusing

Reply:
so this is probably due to my own limitations with pandas, so I'll write here "what I want" and let's see whether there's a way of doing this (/cc @Zaharid)

Right now when the predictions come from libNNPDF they are an object that contain a "central value" and "data" and these are two different quantities.

In vp instead the predictions are a just a pandas dataframe from which one cannot just get the cv because the central value for MC pdfs is not the same central value of hessian PDFs so my solution has been to add a new CV column with the "true" central value.

The problem is that I don't want that extra column to be part of the "data" of the prediction.

Is it clear / is there a solution to this issue?

Reply:
An important constraint I need to include here is that I need the computation of the central value and the envelope to happen at the same time (because it is faster to do fktable x 101 rather than fkatble x 100 and later fktable x 1, so at some point the CV will have to be included in the dataframe).

The way I'm doing that is more or less clear in the example in `class PDF`.
Maybe (probably) the solution is to make a copy of the dataframe at this point so that it doesn't get mutated but it's separated into CV and the envelope of replicas.